maven-java/kalix-maven-plugin/src/main/scala/kalix/RunMojo.scala

```diff
@@ -29,29 +29,30 @@ object RunMojo {
       log.warn("--------------------------------------------------------------------------------------")
     }
 
-    if (jvmArgs.nonEmpty){
+    if (jvmArgs.nonEmpty) {
       log.info("Additional JVM arguments detected: " + jvmArgs.toSeq.mkString(", "))
     }
 
     /*
-     * Collect any sys property starting with `kalix` and rebuild a -D property for each of them
-     * so we can pass it further to the forked process.
+     * Lookup for arguments passed to maven using -Dkey=value and rebuild a -D property for each of them.
      */
-    def collectKalixSysProperties: Seq[String] = {
-      sys.props.collect {
-        case (key, value) if key.startsWith("kalix") => s"-D$key=$value"
-      }.toSeq
-    }
+    def collectSysProperties: Seq[String] =
+      sys.env
+        .get("MAVEN_CMD_LINE_ARGS")
+        .map { args =>
+          args.split("\\s+").toSeq.collect {
+            case arg if arg.startsWith("-D") => arg
+          }
+        }
+        .getOrElse(Seq.empty)
```

Comment:
We were previously picking all properties starting with `kalix` and making it a `-D` property to pass to the underlying process. 

We cannot take all sys properties and pass them down as `-D` args. Instead, we need to read the env var MAVEN_CMD_LINE_ARGS. This contains all the args passed to mvn. We can then take everything that has an explicit `-D` in front of it and consider as an argument to the use application.